provider/system.go

```diff
@@ -0,0 +1,47 @@
+package provider
+
+import (
+	"context"
+	"github.com/ipfs/go-cid"
+)
+
+// System defines the interface for interacting with the value
+// provider system
+type System interface {
+	Run()
+	Close() error
+	Provide(cid.Cid) error
+	Reprovide(context.Context) error
+}
+
+type system struct {
+	provider   Provider
+	reprovider Reprovider
+}
+
+// NewSystem constructs a new provider system from a provider and reprovider
+func NewSystem(provider Provider, reprovider Reprovider) System {
+	return &system{provider, reprovider}
+}
+
+// Run the provider system by running the provider and reprovider
+func (s *system) Run() {
+	go s.provider.Run()
+	go s.reprovider.Run()
+}
+
+// Close the provider and reprovider
+func (s *system) Close() error {
+	// TODO: Close reprovider here
```

Comment:
This doesn't need to be implemented yet because close is a noop on the reprovider at the moment. In the next PR I'm introducing a new reprovider that has close and so will need to be accounted for here.

Reply:
Why not just call it anyways so we won't forget about it?